connect-button/src/main/java/com/ifttt/connect/ui/AnalyticsEventUploader.java

```diff
@@ -0,0 +1,53 @@
+package com.ifttt.connect.ui;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+import java.io.IOException;
+import java.util.List;
+import retrofit2.Response;
+
+/*
+ * Schedules a one time work request to read from the queue, make an api call to submit events and remove them from queue.
+ * */
+public final class AnalyticsEventUploader extends Worker {
+
+    private AnalyticsManager analyticsManager;
+    private static final int MAX_RETRY_COUNT = 3;
+    private static AnalyticsApiHelper apiHelper;
```

Comment:
This should be final. And why does this need to be static?

Reply:
hmm it doesn't, since the `ApiHelper` is singleton.